tiledb/sm/array_schema/axis_schema.h

```diff
@@ -0,0 +1,254 @@
+   * @param capacity Capactiy for the arrays.
+   * @param cell_order The cell order for both arrays.
+   * @param tile_order The tile order for both arrays.
+   */
+  AxisSchema(
+      LabelOrder label_order,
+      shared_ptr<const AxisComponent> index_component,
+      shared_ptr<const AxisComponent> label_component,
+      uint64_t capacity = constants::capacity,
+      Layout cell_order = Layout::ROW_MAJOR,
+      Layout tile_order = Layout::ROW_MAJOR);
+
+  /**
+   * Constructor.
+   *
+   * @param label_order Order of the labels relative to the index for the axis.
+   * @param indexed_array_schema Array schema for the array with indices defined
+   * on the dimension.
+   * @param labelled_array_schema Array schema for the array with labels defined
+   * on the dimension.
+   * @param label_attr_id The integer ID for the label attribute on the indexed
+   * array.
+   * @param index_attr_id The integer ID for the index attribute on the labelled
+   * array.
+   */
+  AxisSchema(
+      LabelOrder label_order,
+      shared_ptr<ArraySchema> indexed_array_schema,
+      shared_ptr<ArraySchema> labelled_array_schema,
+      const attribute_size_type label_attr_id = 0,
+      const attribute_size_type index_attr_id = 0);
+
+  /** Returns the index attribute from the labelled array. */
+  inline attribute_size_type index_attribute_id() const {
+    return index_attr_id_;
+  }
+
+  /** Returns the index attribute from the labelled array. */
+  const Attribute* index_attribute() const;
+
+  /** Returns the index dimension from the indexed array. */
+  const Dimension* index_dimension() const;
+
+  /**
+   * Checks if this axis is compatible as a dimension label for a given
+   * dimension.
+   *
+   * @param dim Dimension to check compatibility against
+   * @returns If the axis is compatible as a dimension label
+   */
+  bool is_compatible_label(const Dimension* dim) const;
+
+  /** Returns the label attribute from the indexed array. */
+  const Attribute* label_attribute() const;
+
+  /** Returns the label dimension from the labelled array. */
+  const Dimension* label_dimension() const;
+
+  /** Returns the index of the label attribute in the indexed array. */
+  inline attribute_size_type label_attribute_id() const {
+    return label_attr_id_;
+  }
+
+  /** Returns the label order type of this axis. */
+  inline LabelOrder label_order() const {
+    return label_order_;
+  }
+
+ private:
+  /** Order of the labels relative to the indices. */
+  LabelOrder label_order_;
+
+  /** Schema for the array with indices defined on the dimension. */
+  shared_ptr<ArraySchema> indexed_array_schema_;
+
+  /** Schema for the array with labels defined on the dimension. */
+  shared_ptr<ArraySchema> labelled_array_schema_;
+
+  /** The integer ID of the label attribute on the indexed array. */
+  attribute_size_type label_attr_id_;
+
+  /** The integer ID of the index attribute on the labelled array. */
+  attribute_size_type index_attr_id_;
```

Comment:
The naming conventions used in this class are confusing. Any suggestions for better names are welcome.